--- a/dlls/ntdll/virtual.c	2018-05-15 12:07:53.839882299 -0400
+++ a/dlls/ntdll/virtual.c	2018-05-16 16:29:00.753042766 -0400
@@ -1387,11 +1387,11 @@ static NTSTATUS map_image( HANDLE hmappi
     server_enter_uninterrupted_section( &csVirtual, &sigset );
 
     if (base >= (char *)address_space_start)  /* make sure the DOS area remains free */
-        status = map_view( &view, base, total_size, mask, FALSE, SEC_IMAGE | SEC_FILE |
+        status = map_view( &view, base, total_size, mask, TRUE, SEC_IMAGE | SEC_FILE |
                            VPROT_COMMITTED | VPROT_READ | VPROT_EXEC | VPROT_WRITECOPY );
 
     if (status != STATUS_SUCCESS)
-        status = map_view( &view, NULL, total_size, mask, FALSE, SEC_IMAGE | SEC_FILE |
+        status = map_view( &view, NULL, total_size, mask, TRUE, SEC_IMAGE | SEC_FILE |
                            VPROT_COMMITTED | VPROT_READ | VPROT_EXEC | VPROT_WRITECOPY );
